From c17e7096d756d35c10dadd50dc38d77dba57e68d Mon Sep 17 00:00:00 2001
From: Steffen Jaeckel <steffen.jaeckel@stzedn.de>
Date: Fri, 22 Mar 2013 13:42:29 +0200
Subject: [PATCH 4/8] include stddef.h per default

---
 libtomcrypt/src/headers/tomcrypt.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libtomcrypt/src/headers/tomcrypt.h b/libtomcrypt/src/headers/tomcrypt.h
index ad27eec..51299af 100644
--- a/libtomcrypt/src/headers/tomcrypt.h
+++ b/libtomcrypt/src/headers/tomcrypt.h
@@ -4,6 +4,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
+#include <stddef.h>
 #include <time.h>
 #include <ctype.h>
 #include <limits.h>
-- 
2.10.1.windows.1

